simplq/src/components/pages/Admin/AdminPage.jsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { useState, useEffect, useCallback } from 'react';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectTokens, selectQueueName } from 'store/selectedQueue';
+import Ribbon from 'components/common/Ribbon';
+import Tour from 'components/common/Tour';
+import { useGetSelectedQueue } from 'store/asyncActions';
+import HeaderSection from './AdminHeaderSection';
+import TokenList from './TokenList';
+import styles from './admin.module.scss';
+import SidePanel from './AdminSidePanel';
+import getToursteps from './TourSteps';
+
+const TIMEOUT = 10000;
+let timeoutId;
+
+export default (props) => {
+  const queueId = props.match.params.queueId;
+  const queueName = useSelector(selectQueueName);
+  const tokens = useSelector(selectTokens);
+  const description = queueName && 'Ready to share';
+  const dispatch = useDispatch();
+  const getSelectedQueue = useGetSelectedQueue();
+
+  const [toursteps, setToursteps] = useState(getToursteps(window.innerHeight));
+  const { isAuthenticated } = useAuth0();
+
+  const update = useCallback(() => {
+    clearTimeout(timeoutId);
+    dispatch(getSelectedQueue({ queueId }));
+    setTimeout(update, TIMEOUT);
+    // TODO: Check if this is good solution.
+    /* eslint-disable-next-line */
+  }, [queueId]);
+
+  const resize = () => setToursteps(getToursteps(window.innerWidth));
+
+  useEffect(() => {
+    window.addEventListener('resize', resize);
+    return () => window.removeEventListener('resize', resize);
+  });
+
+  useEffect(() => {
+    update();
+    return () => clearTimeout(timeoutId);
```

Comment:
@mradenovic here `timeoutId` is not set

Reply:
Best to fix this. Although it doesn't fail, it could cause a memory leak I believe. Later on, we could make something like [this](https://overreacted.io/making-setinterval-declarative-with-react-hooks/#just-show-me-the-code). But for now, let's just fix and use this @mradenovic.

Reply:
Good catch @chalx. Thanks!

> Later on, we could make something like this. But for now, let's just fix and use this @mradenovic.

I like it! Any timer functionality should be moved out of async action (e.g., `../asyncActions/getTotken.jsx`). Then async action can be reused. There is probably a better way to do it than what I did her.